tests/test_daemon.cpp

```diff
@@ -2126,13 +2128,39 @@ TEST_P(DaemonSetExceptions, setHandlesSettingsException)
 }
 
 INSTANTIATE_TEST_SUITE_P(
-    Daemon, DaemonSetExceptions,
-    Values(std::tuple{mp::UnrecognizedSettingException{"foo"}, grpc::StatusCode::INVALID_ARGUMENT,
+    Daemon,
+    DaemonSetExceptions,
+    Values(std::tuple{mp::NonAuthorizedBridgeSettingsException{"reason", "instance", "eth8"},
+                      grpc::StatusCode::INTERNAL,
+                      HasSubstr("Need user authorization to bridge eth8")},
+           std::tuple{mp::UnrecognizedSettingException{"foo"},
+                      grpc::StatusCode::INVALID_ARGUMENT,
                       AllOf(HasSubstr("Unrecognized"), HasSubstr("foo"))},
-           std::tuple{mp::InvalidSettingException{"foo", "bar", "err"}, grpc::StatusCode::INVALID_ARGUMENT,
+           std::tuple{mp::InvalidSettingException{"foo", "bar", "err"},
+                      grpc::StatusCode::INVALID_ARGUMENT,
                       AllOf(HasSubstr("Invalid"), HasSubstr("foo"), HasSubstr("bar"), HasSubstr("err"))},
            std::tuple{std::runtime_error{"Other"}, grpc::StatusCode::INTERNAL, HasSubstr("Other")}));
 
+TEST_F(Daemon, setWorksWhenBridgingThrows)
+{
+    mp::Daemon daemon{config_builder.build()};
+
+    mp::SetRequest request;
+    request.set_key("local.instance.bridged");
+    request.set_val("true");
+
+    const auto& exception = mp::NonAuthorizedBridgeSettingsException{"reason", "instance", "eth8"};
+
+    EXPECT_CALL(mock_settings, set).WillOnce(Throw(exception));
+
+    auto mock_server = StrictMock<mpt::MockServerReaderWriter<mp::SetReply, mp::SetRequest>>{};
+
+    EXPECT_CALL(mock_server, Write(_, _)).WillOnce(Return(true));
+    EXPECT_CALL(mock_server, Read(_)).WillOnce(Return(true));
+
+    auto status = call_daemon_slot(daemon, &mp::Daemon::set, request, mock_server);
```

Comment:
Should you set an expectation on the status as well?

Reply:
Removed this test, see comment below.